staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -412,26 +416,31 @@ func (c *AvailableConditionController) processNextWorkItem() bool {
 	return true
 }
 
-func (c *AvailableConditionController) enqueue(obj *apiregistration.APIService) {
-	key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+func (c *AvailableConditionController) enqueue(apiServiceName string) {
+	c.queue.Add(apiServiceName)
+}
+
+func (c *AvailableConditionController) enqueueService(obj *apiregistration.APIService) {
+	apiServiceName, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
 	if err != nil {
 		klog.Errorf("Couldn't get key for object %#v: %v", obj, err)
 		return
 	}
-
-	c.queue.Add(key)
+	c.queue.Add(apiServiceName)
 }
 
 func (c *AvailableConditionController) addAPIService(obj interface{}) {
 	castObj := obj.(*apiregistration.APIService)
 	klog.V(4).Infof("Adding %s", castObj.Name)
-	c.enqueue(castObj)
+	c.updateCacheForAPIService(castObj)
+	c.enqueueService(castObj)
 }
 
 func (c *AvailableConditionController) updateAPIService(obj, _ interface{}) {
 	castObj := obj.(*apiregistration.APIService)
 	klog.V(4).Infof("Updating %s", castObj.Name)
-	c.enqueue(castObj)
+	c.updateCacheForAPIService(castObj)
```

Comment:
only need to call `rebuildAPIServiceCache` if `(castObj.Spec.Service != nil || oldCastObj.Spec.Service != nil) && !reflect.DeepEqual(castObj.Spec.Service, oldCastObj.Spec.Service)`